frontend/src/metabase/status/hooks/use-check-active-downloads-before-unload.ts

```diff
@@ -0,0 +1,13 @@
+import { useBeforeUnload } from "react-use";
+import { t } from "ttag";
+
+import { useSelector } from "metabase/lib/redux";
+import { hasActiveDownloads } from "metabase/redux/downloads";
+
+export const useCheckActiveDownloadsBeforeUnload = () => {
+  const downloadInProgress = useSelector(hasActiveDownloads);
+  useBeforeUnload(
+    downloadInProgress,
+    t`Export in progress. Are you sure you want to leave?`,
```

Comment:
This message does not appear for me, I just get the browser's generic `Changes you made may not be saved.` message.

![Screenshot 2024-07-25 at 9.14.17 AM.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/9ULpbAzGBtvQRyZEpGvu/8cf28465-18f9-4b16-b946-696cec54e44b.png)

Reply:
It depends on a browser. Modern browsers tend to remove displaying custom messages